### PR TITLE
Enables MSAN for Suricata

### PR DESCRIPTION
**Pull request by catenacyber**
Will work when https://github.com/OISF/suricata/pull/5422 gets merged

### CODE REVIEW

File: projects/suricata/build.sh [hunk 1]
```diff
@@ -45,6 +45,7 @@ make install
 cd ..
 
 export CARGO_BUILD_TARGET="x86_64-unknown-linux-gnu"
+rustup component add rust-src --toolchain nightly
```
Comment (catenacyber) [inline projects/suricata/build.sh:48]: You may want to do this in the Docker image, so that it works for other Rust projects

Comment (inferno-chromium) [reply projects/suricata/build.sh:48]: Can you submit a patch to https://github.com/google/oss-fuzz/blob/master/infra/base-images/base-builder/Dockerfile#L79 with a comment on why it is needed. Thanks!

**Review by inferno-chromium — COMMENTED** [review]
Can you check why msan build is still failing

**Comment by catenacyber** [comment]
MSAN build is failing because we need Suricata upstream merge of OISF/suricata#5422

File: projects/suricata/build.sh (hunk 1 above)
Comment (catenacyber) [reply projects/suricata/build.sh]: It comes from this discussion :
https://github.com/google/oss-fuzz/issues/3469#issuecomment-648463139

We need to have `-Z build-std` to rebuild Rust standard library with MSAN

And to have this to work, we need a few things including this `rustup component add rust-src` cf 
https://doc.rust-lang.org/cargo/reference/unstable.html#build-std

**Comment by catenacyber** [comment]
Ping @inferno-chromium 
Can we run CI again ?

**Comment by jonathanmetzman** [comment]
> Ping @inferno-chromium
> Can we run CI again ?

Tip: You can force this yourself by pushing to this PR.

**Comment by catenacyber** [comment]
Thanks Johnathan, I rebased and force-pushed

**Comment by catenacyber** [comment]
Does the CI take into account the changes to base-builder ?

**Comment by jonathanmetzman** [comment]
> Does the CI take into account the changes to base-builder ?

I think so?

**Comment by catenacyber** [comment]
If the CI succeeds with this new commit, I guess CI was wrong last time

**Comment by inferno-chromium** [comment]
> Does the CI take into account the changes to base-builder ?

I dont think so. But this should work, let me remove your dummy thing and commit. keep an eye on build tmrw.

**Comment by jonathanmetzman** [comment]
> Does the CI take into account the changes to base-builder ?

Maybe I misunderstood what you meant. Did you make changes in this PR?
Then CI not taking this into account is expected.

**Comment by catenacyber** [comment]
Looks like it is working :-)
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=26081

**Pull request merged**